src/librustdoc/html/highlight.rs

```diff
@@ -73,14 +109,14 @@ enum Class {
     KeyWord,
     // Keywords that do pointer/reference stuff.
     RefKeyWord,
-    Self_,
+    Self_(LightSpan),
```

Comment:
It's a bit weird to add the span to `Class`. I think it would make more sense to add it as a field to `Highlight::Token` and leave `Class` alone.

Reply:
I'm a bit shared on this: currently, only the classes with the need for the `LightSpan` have one, meaning it cannot be forgotten because the code impose it. If I move it into the `Token` enum, I either give it for all classes, even those which don't need it and add an extra argument to the `string` function or I make it an optional field but then it's up to the programmer to not forget to set this field.